src/app/common/resolvers/search.resolver.ts

```diff
@@ -9,10 +9,28 @@ import { inject } from "@angular/core";
 import { ActivatedRouteSnapshot, ResolveFn } from "@angular/router";
 import ProjectLinks from "src/app/project-links";
 import { SearchService } from "src/app/search/search.service";
+import { Observable, of, switchMap } from "rxjs";
+import { DatasetSearchResult } from "src/app/interface/search.interface";
+import { LoggedUserService } from "src/app/auth/logged-user.service";
+import { AppConfigService } from "src/app/app-config.service";
 
-export const searchResolver: ResolveFn<void> = (route: ActivatedRouteSnapshot) => {
+export const searchResolver: ResolveFn<Observable<DatasetSearchResult>> = (route: ActivatedRouteSnapshot) => {
     const searchService = inject(SearchService);
+    const loggedUserService = inject(LoggedUserService);
+    const appConfigService = inject(AppConfigService);
+    const isAuthenticated = loggedUserService.isAuthenticated;
+    const isSemanticSearchAvailable = appConfigService.semanticSearchScore;
     const page = route.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_PAGE) ?? 1;
     const query = route.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_QUERY) ?? "";
-    return searchService.searchDatasets(query, Number(page) - 1);
+
+    return searchService.searchDatasets(query, Number(page) - 1).pipe(
+        switchMap((result: DatasetSearchResult) => {
+            return result.datasets.length ||
+                !query ||
+                !isAuthenticated ||
+                typeof isSemanticSearchAvailable === "undefined"
+                ? of(result)
+                : searchService.semanticSearchDatasets(query);
+        }),
```

Comment:
The logic is not clear enough. Try to simpify it and maybe add some examples in comment.

"If I have results from default method, or query is empty, or I'm not authenticated, or semantic search is not enabled, I get default method results, otherwise, I'm redirected to to semantic search."

Why authentication is a condition? 
If semantic search really requires it, I'm a bit surprised, but if that is true, can you take this into account when assigning "isSemanticSearchAvailable" initially.

Reply:
Fixed